tests/auth.rs

```diff
@@ -43,11 +43,15 @@ fn auth_works(
     sleep(Duration::from_secs(1));
 
     let client = reqwest::Client::new();
-    let body = client
+    let response = client
         .get(format!("http://localhost:{}", port).as_str())
         .basic_auth(client_username, Some(client_password))
-        .send()?
-        .error_for_status()?;
+        .send()?;
+
+    let status_code = response.status();
+    assert_eq!(status_code.canonical_reason(), Some("OK"));
```

Comment:
I like this generally, but I'd prefer if we could do it like [here](https://github.com/svenstaro/miniserve/pull/102#discussion_r281189844).